src/js/game/building_codes.js

```diff
@@ -56,40 +50,21 @@ export function registerBuildingVariant(
 
 /**
  *
- * @param {number} code
+ * @param {String} code
  * @returns {BuildingVariantIdentifier}
  */
 export function getBuildingDataFromCode(code) {
     assert(gBuildingVariants[code], "Invalid building code: " + code);
     return gBuildingVariants[code];
 }
 
-/**
- * Builds the cache for the codes
- */
-export function buildBuildingCodeCache() {
-    for (const code in gBuildingVariants) {
-        const data = gBuildingVariants[code];
-        const hash = data.metaInstance.getId() + "/" + data.variant + "/" + data.rotationVariant;
-        variantsCache.set(hash, +code);
-    }
-}
-
 /**
  * Finds the code for a given variant
  * @param {MetaBuilding} metaBuilding
  * @param {string} variant
  * @param {number} rotationVariant
- * @returns {number}
+ * @returns {String}
  */
-export function getCodeFromBuildingData(metaBuilding, variant, rotationVariant) {
-    const hash = metaBuilding.getId() + "/" + variant + "/" + rotationVariant;
-    const result = variantsCache.get(hash);
-    if (G_IS_DEV) {
-        if (!result) {
-            console.warn("Known hashes:", Array.from(variantsCache.keys()));
-            assertAlways(false, "Building not found by data: " + hash);
-        }
-    }
-    return result;
+export function getCodeFromBuildingData(metaBuilding, variant, rotationVariant = 0) {
+    return metaBuilding.getId() + "/" + variant + "/" + rotationVariant;

```

Comment:
this is very inefficent, since it is generating a long string. it also bloats up the savegame by a lot. have to find a better solution for this somehow.

Maybe the builtin buildings could have an integer code, the mod buildings a string?

Reply:
It can all be a number. The problem is that when you load a world with different mods different buildings are registered to the numbers

Reply:
For 10.000 buildings it's about 0.23mb difference

Reply:
I tested with some worlds:
One went from 11.4mb to 11.5mb
And another went from 3.70mb to 3.71mb

Reply:
is that the compressed version?

Reply:
The bin files

Reply:
the bin files are 11 mb?

Reply:
Yes and it has about 200000 buildings